### ISSUE TITLE
Dependabot fails with Rust named profiles

### ISSUE THREAD

**Issue by Guara92** [dependabot/dependabot-core#4676]
Dependabot fails to parse `cargo.toml` with named profiles due to outdated Cargo version
![image](https://user-images.githubusercontent.com/12493527/151345966-02a2c33d-a4fa-4b4d-907d-4a26b9a7d36f.png)
**Package ecosystem**
Rust

**Package manager version**
Cargo 1.58.0

**Language version**
Rust 1.58.1

**Profile Example**
```toml
[profile.release]
lto = "off"
opt-level = 2

[profile.production]
inherits = "release"
lto = "thin"
codegen-units = 8
opt-level = 3
incremental = false
```

**Comment by Nishnha** [comment]
Hi @Guara92,

We currently do not support Cargo v1.58.0, which is likely why you're seeing the `named-profiles` error you linked in the screenshot.

Once we update Cargo to 1.58.0 it should resolve that issue, but I can't offer a timeline on when that will be; we ran into an issue when we updated Dependabot from `Cargo v1.51` to `Cargo v1.57`. [We reverted the version bump](https://github.com/dependabot/dependabot-core/pull/4588) and are now working to identify the root issue.
